src/xml_parser_main_metadata_together.c

```diff
@@ -454,6 +455,10 @@ int cr_xml_parse_main_metadata_together(const char *primary_path,
         }
     }
 
+    cr_package_free(primary_pd->pkg);
+    cr_package_free(filelists_pd->pkg);
+    cr_package_free(other_pd->pkg);
```

Comment:
Does this need the same checks as e.g. https://github.com/rpm-software-management/createrepo_c/blob/master/src/xml_parser_filelists.c#L364-L372